trappy/devfreq_power.py

```diff
@@ -72,4 +72,76 @@ def get_all_freqs(self):
 
         return pd.DataFrame(self.data_frame["freq"] / 1000000)
 
+    def plot_cdev_states(self, title="", width=None, height=None, xlim="default",
+                         ylim="range", legend_label="", devices=None):
+        """Plot the cooling device state evolution
+
+        :param ax: Axis instance
+        :type ax: :mod:`matplotlib.Axis`
+
+        :param legend_label: Label for the legend
+        :type legend_label: str
+
+        :param title: The title of the plot
+        :type title: str
+
+        :param width: The width of the plot
+        :type width: int
+
+        :param height: The height of the plot
+        :type height: int
+
+        :param devices: List of devfreq devices to plot
+            All are plotted by default
+        :type tz_id: list of str
+        """
+        from matplotlib import pyplot as plt
+        from trappy.plot_utils import normalize_title, pre_plot_setup, post_plot_setup
+
+        if len(self.data_frame) == 0:
+            raise ValueError("Empty DataFrame")
+
+        thermal_dfr = self.data_frame.copy()
+
+        # Find available devices
+        available_devices = []
+        for device in thermal_dfr["type"].unique().tolist():
+            available_devices.append(device)
+
+        # Sanitize devices
+        if devices is not None:
+
+            thermal_dfr = thermal_dfr[
+                thermal_dfr["type"].isin(devices)
+            ]
+
+            if len(thermal_dfr) == 0:
+                raise ValueError("No {} trace for devices {}".format(self.unique_word, devices))
+
+        cdevs = {}
+        # At this point we've removed the devices we don't care about
+        for device in thermal_dfr["type"].unique().tolist():
+            cdevs[device] = thermal_dfr[thermal_dfr["type"] == device]["cdev_state"]
+
+        # Prepare one subplot per cdev
+        ax = pre_plot_setup(width, height, nrows = len(cdevs))
+
+        several_graphs = len(cdevs) > 1
+
+        for idx, (device, df) in enumerate(cdevs.iteritems()):
+            _ax = ax[idx] if several_graphs else ax
+
+            cdev_title = "{}".format(device)
+            cdev_label = normalize_title("cdev_state", legend_label)
+            df.plot(ax=_ax, title=cdev_title, drawstyle="steps-post")
+
+            _ax.legend([cdev_label])
+            # Add vertical guidelines
+            _ax.yaxis.grid(True)
+            # Ensure tick interval is 1 and not something with weird decimals
+            states = df.unique().tolist()
+            _ax.set_yticks(range(min(states), max(states) + 1))
+
+            post_plot_setup(_ax, xlim=xlim, ylim=ylim)
+
```

Comment:
Would be nice to avoid the duplicated code. If it's a big effort to do that then I think we can live with it, though.